pkg/apis/crd/v1beta1/types.go

```diff
@@ -210,6 +210,8 @@ type ExternalIPPool struct {
 type ExternalIPPoolSpec struct {
 	// The IP ranges of this IP pool, e.g. 10.10.0.0/24, 10.10.10.2-10.10.10.20, 10.10.10.30-10.10.10.30.
 	IPRanges []IPRange `json:"ipRanges"`
+	// The Subnet info of this IP pool. If set, all IP ranges in the IP pool should share the same subnet attributes.
+	SubnetInfo *SubnetInfo `json:"subnetInfo,omitempty"`
```

Comment:
Since ExternalIPPool is used by other features besides Egress (at least by `ServiceExternalIP`), will this field ever be relevant for these other features? For example, VLAN tagging / untagging for external Service traffic?

Should the documentation mention that this field is only used when an IP is allocated from the pool for the Egress feature, and is ignored otherwise?

Reply:
I haven't verified whether it can be used by ServiceExternalIP but likely infeasible because ServiceExternalIP requires the IP to be not assigned to the Node physically, which can't work with VLAN sub-interface according to my experiments when implementing this feature.

I have added a comment to the field about this.

Reply:
Do you think better to reuse `SubnetIPRange` to be consistent with `IPPool`?

Reply:
I considered this in the beginning but found some deficiencies in this way:
1. With a list of SubnetIPRange in a single pool, it‘s possible to have different subnets in a pool, which makes implementation complex and inefficient: for example, when we want to know the subnet info of an IP, we have to loop over all ranges to figure it out; whenever there is a change in any of the IP ranges, we have to resync all resources associated with IPs allocated from it, increasing the span of the a single pool.
2. It may be unclear to users that they should create a pool for each subnet or a pool for all subnets. From API's perspective, one object for one resource is clearer and simpler.
3. It's redundant to fill the same subnet info when multiple ip ranges sharing the subnet.

I talked about this in the community meeting https://www.youtube.com/watch?v=q3vaVzzH6WM, starting from 20:42.

Reply:
With your points, should we update IPPool to have a single `SubnetInfo`?

Reply:
Yes, I discussed with @gran-vmv, I think there was no specific reason why IPPool was made so, we can update it when bumping it up to beta.